packages/smooth_app/lib/pages/product/common/product_list_popup_items.dart

```diff
@@ -216,6 +225,135 @@ class ProductListPopupOpenInWeb extends ProductListPopupItem {
   }
 }
 
+/// Popup menu item for the product list page: export list.
+class ProductListPopupExport extends ProductListPopupItem {
+  @override
+  String getTitle(final AppLocalizations appLocalizations) =>
+      appLocalizations.product_list_export;
+
+  @override
+  IconData getIconData() => Icons.download;
+
+  @override
+  ProductListPopupMenuEntry getEntry() => ProductListPopupMenuEntry.export;
+
+  @override
+  bool isDestructive() => false;
+
+  @override
+  Future<ProductList?> doSomething({
+    required final ProductList productList,
+    required final LocalDatabase localDatabase,
+    required final BuildContext context,
+  }) async {
+    final String listName = ProductQueryPageHelper.getProductListLabel(
+      productList,
+      AppLocalizations.of(context),
+    );
+    final String fileName = _buildFileName(listName);
+    final StringBuffer csv = StringBuffer(
+        'Barcode${ProductListPopupItem.EXPORT_SEPARATOR}Name${ProductListPopupItem.EXPORT_SEPARATOR}Brand\n');
+
+    for (final String barcode in productList.barcodes) {
+      final Product? product = await DaoProduct(localDatabase).get(barcode);
+      if (product != null) {
+        final String name = (product.productName ?? '')
+            .replaceAll(ProductListPopupItem.EXPORT_SEPARATOR, '');
+        final String brand = (product.brands ?? '')
+            .replaceAll(ProductListPopupItem.EXPORT_SEPARATOR, '');
+        csv.write(
+            '"$barcode"${ProductListPopupItem.EXPORT_SEPARATOR}"$name"${ProductListPopupItem.EXPORT_SEPARATOR}"$brand"\n');
```

Comment:
The same goes for here.
+ you have the [writeln](https://api.dart.dev/dart-core/StringBuffer/writeln.html) method